lib/data/api/http_client.dart

```diff
@@ -21,7 +21,11 @@ class HttpClient {
       response = await get(path);
       final statusCode = response.statusCode;
       if (statusCode >= 200 && statusCode < 299) {
-        return jsonDecode(response.body);
+        if (statusCode == 204) {
+          throw EmptyResultException();
```

Comment:
Chyba coś nie tak, jeśli jest 204 to nie powinien lecieć wyjątek bezwarunkowo.

Reply:
jest sprawdzane w repo czy lista nie jest pusta, więc wywaliłem to

Reply:
Chodziło mi o to, że przy 204 body jest puste lub nie istnieje. 
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204
Nie wiem jak dokładnie zachowuje się ta flutterowa libka do http, ale kiedyś Retrofit miał z tym problem np. https://stackoverflow.com/questions/27940277/retrofit-callback-for-204-no-content-response
Jeśli wystawia nulla przy 204 to jsonDecode się może wywalić (chyba że akceptuje nulle, nie sprawdzałem).

Reply:
sprawdziłem to i przy 204 jak nie body jest puste to `jsonDecode` wyrzuca format exception. Zrobiłem tak, że w takiej sytuacji zamiast to decodować to zwracam pustą listę i poszczególne metody repo już sobie to obsłużą jak chcą